### ISSUE TITLE
Ability to run gabbi test cases individually

### ISSUE THREAD

**Issue by dkt26111** [cdent/gabbi#220]
The ability to run an individual gabbi test without any of the tests preceding it in the yaml file could be useful.  I created a project where I drive gabbi test cases using Robot Framework (https://github.com/dkt26111/robotframework-gabbilibrary).  In order for that to work I explicitly set the prior field of the gabbi test case being run to None.

**Comment by cdent** [comment]
Wow, this is very interesting. Thanks for pointing out your work.

Did you consider putting each individual test in a separate YAML file? That would be one way of overcoming the issue you're having with prior.

If we did add something to support this in the core of gabbi, are you thinking that it would be best implemented as a flag passed to `build_tests` that effectively says "don't set prior"?

**Comment by dkt26111** [comment]
I did consider using separate YAML files but I think in any real world project the number of files would proliferate quickly and become unmanageable.  Also I like the idea of keeping related API test cases in the same YAML file.  That seemed to work well for my sample API tests (https://github.com/dkt26111/robotgabbi-openstack/blob/master/gabbits/nova.yaml)

A flag to `build_tests` would be perfect.

**Comment by cdent** [comment]
@FND @jd @jasonamyers any opinions on this?

**Comment by jasonamyers** [comment]
I agree with @dkt26111 about the keeping the tests grouped together in a file. I think a flat to build_tests is perfectly reasonable.

**Comment by cdent** [comment]
@dkt26111 is this a feature you'd be able to create a pull request for? If so we could review it pretty quickly and get it merged. Otherwise, I can do it myself but it might not be done as soon.

**Comment by dkt26111** [comment]
No problem.  I will make the changes and submit a pull request for it.

**Comment by jd** [comment]
I can see the usefulness of that, though to me that would be some special kind of YAML files that you would mark as such in the YAML file itself. Like "those tests in the YAML file can be run independently in any order". Does that make sense?

**Comment by cdent** [comment]
> such in the YAML file itself

You mean something in the yaml separate from the tests key like:

```yaml
use_prior: False
tests:
- name: [...]
```

That might be useful too, but the impression I'm getting from reading @dkt26111's blog post and the related github projects is that one of the ideas is to be able to take an existing corpus of gabbi tests and farm them, unmodified, for use in this other framework.

Is that the case?

**Comment by jd** [comment]
@cdent exactly what I mean

I don't think you can "farm" any existing scenario file if they are written as a serie of request. That won't work so it sounds like a weird project.

**Comment by dkt26111** [comment]
I like the idea of the use_prior flag inside the YAML file.  That way it is obvious from looking at the YAML file how the tests will behave.

@cdent As far as my project is concerned, I don't have any requirement to use unmodified YAML files.  Since I control the YAML files for my tests, I'm free to modify it.

Should the use_prior flag in the YAML file be inside the `defaults` section so that it can be overridden by each test case if necessary:

```yaml
defaults:
    use_prior: False

tests:
- name: foo
  ...

- name: bar
  use_prior: True
  ...
```

For my purposes it doesn't matter if it's inside the `defaults` section or outside since I will probably only set the `use_prior` flag to apply to the entire YAML file.

**Comment by cdent** [comment]
For reference there's vast amounts of meandering irc conversation about this today: http://p.anticdent.org/2eKx

**Comment by cdent** [comment]
The idea of making this test level (and thus available via `defaults`) is probably most in keeping with the model that gabbi has, so yeah, very good idea. It also makes it relatively easy to implement.

I think it would still make sense to be able to pass a parameter to `build_tests` in case changing existing yaml files was not a good option. There's an example of this at https://github.com/cdent/gabbi/blob/9cfbff2bd616bccc9349effa9967a427ba44679c/gabbi/runner.py#L116-L134 that might be useful.

**Comment by dkt26111** [comment]
I made the changes in my forked repo:
https://github.com/dkt26111/gabbi/commit/9a5bb0cb6e83881c3084102ff09626a812236207

* I'm using the name `ordered_test_sequence` instead of `use_prior`.  Still not quite sure if that's the best name for the directive/flag.
* I didn't modify gabbi/runner.py to set the parameter for `ordered_test_sequence`.  I don't think it makes sense to allow the user to change this from the command line.
* I manually tested this change against my YAML files and it seems to be working.  But I still need to add unittests for this new directive and flag.

Let me know what you think so far.
